### DISCUSSION TITLE
How to trigger datagrid reset manually

### DISCUSSION THREAD

**Discussion by ph55** [vmware/clarity#5922] [Q&A] [answered]
I'm trying to make pagination configurable. So it could be turned on and off.
But after turning pagination off datagrid not getting rebuilt (should show 100 rows).

Pagination on:
![image](https://user-images.githubusercontent.com/1386622/116791909-135c6180-aac6-11eb-880e-1e8ded0e3287.png)

Pagination off (should display 100 rows):
![image](https://user-images.githubusercontent.com/1386622/116791933-3e46b580-aac6-11eb-8810-ac3d3257b1a3.png)

Is there any way to trigger reset manually?
I've tried `dataChanged` and `refresh.next` - both didn't work.

Code below:
```ts
import { Component, OnInit, ViewChild } from '@angular/core';
import { ClrDatagrid } from '@clr/angular';

@Component({
  selector: 'app-test',
  template: `
    <button class="btn" (click)="togglePagination()">Toggle Pagination</button>

    <clr-datagrid>
      <clr-dg-column>ID</clr-dg-column>
      <clr-dg-column>Score</clr-dg-column>

      <clr-dg-row *clrDgItems="let score of scores">
        <clr-dg-cell>{{ score.id }}</clr-dg-cell>
        <clr-dg-cell>{{ score.score }}</clr-dg-cell>
      </clr-dg-row>

      <clr-dg-footer>
        <clr-dg-pagination
          *ngIf="paginationEnabled"
          [clrDgPageSize]="5"
          [clrDgTotalItems]="scores.length"
        >
          <clr-dg-page-size [clrPageSizeOptions]="[5, 10, 20, 50, 100]">
            Rows per page:
          </clr-dg-page-size>
          {{ scores.length }} total
        </clr-dg-pagination>
      </clr-dg-footer>
    </clr-datagrid>
  `,
})
export class TestComponent implements OnInit {
  paginationEnabled = true;
  scores: { id: number; score: any }[] = [];

  @ViewChild(ClrDatagrid) dg!: ClrDatagrid;

  constructor() {}

  ngOnInit(): void {
    while (this.scores.length < 100) {
      this.scores.push({ id: this.scores.length + 1, score: Math.random() });
    }
  }

  togglePagination(): void {
    this.paginationEnabled = !this.paginationEnabled;

    this.dg.dataChanged(); // not working
    this.dg.refresh.next(); // not working
  }
}

```

**Comment by Jinnie** [comment, marked as answer]
We don't have template API for this, so it's not really straightforward, but it's achievable on typescript API level by following these steps:
- taking a reference of the ClrDatagridPagination
- setting page size to 0
- deactivating the pagination
- calling dg.dataChanged() now
Also in the API you need to keep the paging component alive (or you'll lose the reference to the page service), thus you have to move the `*ngIf` to the `clr-dg-page-size` element, which is obsolete if paging is disabled.
Here is the example:
https://stackblitz.com/edit/pattern-toggle-dg-paging?file=src/app/app.component.ts

**Reply by Jinnie** [reply]
The above steps should be considered a workaround.
A real fix would be for us to try cleaning up the Page service in ClrDatagridPaging#ngOnDestroy().
We're resetting the page size there, but not resetting `activated` to false.
